code/modules/ghostroles/instantiator.dm

```diff
@@ -155,21 +155,35 @@
 	var/equip_loadout = TRUE
 	/// equip traits
 	var/equip_traits = TRUE
+	/// Allows the selection of specific species for ghost roles.
+	var/species_required = null
+	/// Allows the blacklisting of specific species from ghost roles.
+	//var/species_restricted = null
 
 /datum/ghostrole_instantiator/human/player_static/Create(client/C, atom/location, list/params)
 	var/mob/living/carbon/human/H = ..()
 	var/list/errors = list()
 	// todo: respect warnings; we ignore them right now so we don't block joins.
-	if(!C.prefs.spawn_checks(PREF_COPY_TO_FOR_GHOSTROLE, errors))
+	if(!C.prefs.spawn_checks(PREF_COPY_TO_FOR_GHOSTROLE | PREF_COPY_TO_NO_CHECK_SPECIES, errors))
 		to_chat(C, SPAN_WARNING("<h3><center>--- Character Setup Errors - Please resolve these to continue ---</center></h3><br><b>-&nbsp;&nbsp;&nbsp;&nbsp;[jointext(errors, "<br>-&nbsp;&nbsp;&nbsp;&nbsp;")]</b>"))
 		return
 
+	if(!isnull(species_required) && species_required != C.prefs.real_species_datum().type)
+		to_chat(C, SPAN_WARNING("<h3><center>--- Character Species Is Not Allowed In This Role - Please resolve these to continue ---</center></h3><br><b>-&nbsp;&nbsp;&nbsp;&nbsp;[jointext(errors, "<br>-&nbsp;&nbsp;&nbsp;&nbsp;")]</b>"))
+		return
+
+	/*
+	if(!isnull(species_restricted) && species_restricted == C.prefs.real_species_datum().type)
+		to_chat(C, SPAN_WARNING("<h3><center>--- Character Species Is Not Allowed In This Role - Please resolve these to continue ---</center></h3><br><b>-&nbsp;&nbsp;&nbsp;&nbsp;[jointext(errors, "<br>-&nbsp;&nbsp;&nbsp;&nbsp;")]</b>"))
+		return
+	*/
+
 	LoadSavefile(C, H)
 	return H
 
 /datum/ghostrole_instantiator/human/player_static/proc/LoadSavefile(client/C, mob/living/carbon/human/H)
 	C.prefs.copy_to(H)
-	SSjob.EquipRank(H, USELESS_JOB)
+	//SSjob.EquipRank(H, USELESS_JOB)
```

Comment:
please dont use comments to disable no longer needed code. Delete it or keep it active
Same goes for the block comment above

Reply:
ez